lib/netext/httpext/transport.go

```diff
@@ -115,9 +115,17 @@ func (t *transport) measureAndEmitMetrics(unfReq *unfinishedRequest) *finishedRe
 			tags["status"] = "0"
 		}
 	} else {
+		cleanURL := URL{u: unfReq.request.URL, URL: unfReq.request.URL.String()}.Clean()
 		if enabledTags.Has(stats.TagURL) {
-			u := URL{u: unfReq.request.URL, URL: unfReq.request.URL.String()}
-			tags["url"] = u.Clean()
+			tags["url"] = cleanURL
+		}
+
+		if _, ok := tags["name"]; !ok && enabledTags.Has(stats.TagName) {
+			tags["name"] = cleanURL
+		}
```

Comment:
I would've really prefered it, if cleanURL is created only if it's needed :D

Reply:
I changed this in 8872f4ab, at the expense of duplication, though I'm not sure it will be more efficient overall, as most of the time both `url` and `name` tags will be provided. We can add additional checks, but I don't want to complicate this too much.

Reply:
Changed this to check all redirects in e6b65c87.